mailserver/mailserver_db_postgres.go

```diff
@@ -79,26 +82,36 @@ func (i *PostgresDB) BuildIterator(query CursorQuery) (Iterator, error) {
 		// We disable security checks as we need to use string interpolation
 		// for this, but it's converted to 0s and 1s so no injection should be possible
 		/* #nosec */
-		stmtString = fmt.Sprintf("SELECT id, data FROM envelopes where id >= $1 AND id < $2 AND bloom & b'%s'::bit(512) = bloom ORDER BY ID DESC LIMIT $3", toBitString(query.bloom))
+		stmtString = fmt.Sprintf(`
+			SELECT id, data 
+			FROM envelopes 
+			WHERE id >= $1 AND id < $2 AND (bloom & b'%s'::bit(512) = bloom OR topic = any($3)) 
```

Comment:
I would split the two queries, unless there's an actual use case for specifying both `bloom` and `topic`, reason being that `OR` queries are tricky to index, and we might get degraded performance.

Say something like:
```
if len(topics) > 0 {
  // perform topic query 
} else {
  // perform bloom query
}
```

this way we can create an index on `id` and topics if it does not exists already.